staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +50,194 @@ func TestDecode(t *testing.T) {
 		want          interface{}
 		assertOnError func(t *testing.T, e error)
 	}{
+		{
+			name: "reject duplicate keys (int)",
+			in:   hex("a201010102"), // {1:1, 1:2}
+			into: map[int64]interface{}{},
```

Comment:
Decoding into struct{} might be a more valuable test. We know there is library support for mapping int keys to fields, and we should reject duplicates there even if that support isn't being used.